src/admin/api.js

```diff
@@ -349,6 +353,88 @@ function extractDestinationCurrency(currencyPair) {
     return currencyPair.substring(3, 6);
 }
 
+/**
+ * Generates the required object structure for a ForEx provider.
+ * @function getForexProviderInfo
+ * @param {String} forexProviderName
+ * @param {String} currencyPair
+ * @param {Object} rateDetails
+ * @returns {Object}
+ */
+function getForexProviderInfo(forexProviderName, currencyPair, rateDetails) {
+    const sourceCurrency = extractSourceCurrency(currencyPair);
+
+    function getBidSpotRate(rate, decimalPlaces) {
```

Comment:
Pull this out to top level so that it can be tested and reused.

Reply:
Done.